lib/corpus.py

```diff
@@ -390,6 +390,19 @@ def __init__(
 
         self.recording = recording
 
+    def full_orth(self) -> str:
+        """
+        :return: Left context orth (if any) + orth + right context orth (if any).
+        """
+        text = ""
+        if self.left_context_orth is not None:
+            text += self.left_context_orth.strip() + " "
+        text += self.orth
+        if self.right_context_orth is not None:
+            text += " " + self.right_context_orth.strip()
+
+        return text
```

Comment:
Can there be a case where `self.orth == ""`?
```suggestion
        return " ".join([s for s in [self.left_context_orth, self.orth, self.right_context_orth] if s])
```

Reply:
Sure, orth can even be None.

Reply:
> Can there be a case where self.orth == ""?

There can definitely be such a case, it totally depends on the user. Although it's not normal, and implies that there might be something wrong with the segment (e.g. TN removed all of the data).

Your code is more dense and meaningful, I'll use your approach, which also works nicely when everything's `None`.